pkg/operator/controller/gatewayapi/status.go

```diff
@@ -0,0 +1,80 @@
+)
+
+// setUnmanagedGatewayAPICRDNamesStatus sets the status of the ingress cluster operator
+// with the names of the unmanaged Gateway CRDs.
+func (r *reconciler) setUnmanagedGatewayAPICRDNamesStatus(ctx context.Context, crdNames []string) error {
+	return r.setClusterOperatorStatusExtension(ctx, &status.IngressOperatorStatusExtension{
+		UnmanagedGatewayAPICRDNames: strings.Join(crdNames, ","),
+	})
+}
+
+// setClusterOperatorStatusExtension attempts to ensure that the ingress cluster operator's status
+// is updated with the given extension. Returns an error if failed to update the status.
+func (r *reconciler) setClusterOperatorStatusExtension(ctx context.Context, desiredExtension *status.IngressOperatorStatusExtension) error {
+	have, current, err := r.currentClusterOperator(ctx, operatorcontroller.IngressClusterOperatorName())
+	if err != nil {
+		return err
+	}
+	if !have {
+		return fmt.Errorf("cluster operator %q not found", operatorcontroller.IngressClusterOperatorName().Name)
+	}
+	if _, err := r.updateClusterOperatorStatusExtension(ctx, current, desiredExtension); err != nil {
+		return err
+	}
+	return nil
+}
+
+// currentClusterOperator returns a boolean indicating whether a cluster operator
+// with the given name exists, as well as its definition if it does exist and an error value.
+func (r *reconciler) currentClusterOperator(ctx context.Context, name types.NamespacedName) (bool, *configv1.ClusterOperator, error) {
+	co := &configv1.ClusterOperator{}
+	if err := r.client.Get(ctx, name, co); err != nil {
+		if errors.IsNotFound(err) {
+			return false, nil, nil
+		}
+		return false, nil, fmt.Errorf("failed to get cluster operator %q: %w", name.Name, err)
+	}
+	return true, co, nil
+}
+
+// updateClusterOperatorStatusExtension updates a cluster operator's status extension.
+// Returns a boolean indicating whether the cluster operator was updated, and an error value.
+func (r *reconciler) updateClusterOperatorStatusExtension(ctx context.Context, current *configv1.ClusterOperator, desiredExtension *status.IngressOperatorStatusExtension) (bool, error) {
+	currentExtension := &status.IngressOperatorStatusExtension{}
+	if len(current.Status.Extension.Raw) > 0 /*to avoid "unexpected end of JSON input" error*/ {
+		if err := json.Unmarshal(current.Status.Extension.Raw, currentExtension); err != nil {
+			return false, fmt.Errorf("failed to unmarshal status extension of cluster operator %q: %w", current.Name, err)
+		}
+	}
+	if equality.Semantic.DeepEqual(*currentExtension, *desiredExtension) {
+		return false, nil
+	}
+
+	updated := current.DeepCopy()
+	rawDesiredExtension, err := json.Marshal(desiredExtension)
+	if err != nil {
+		return false, fmt.Errorf("failed to marshal desired status extension of cluster operator %q: %w", updated.Name, err)
+	}
+	updated.Status.Extension.Raw = rawDesiredExtension
+	if err := r.client.Status().Update(ctx, updated); err != nil {
```

Comment:
This stomps any unrecognized data in `status.extension`.  If another controller needed to store some state in `status.extension`, this logic would be problematic.  Could you add a comment to that effect (possibly a "to do" comment to preserve unrecognized data)?

Reply:
If you plan to make any other changes, please do check if there's data in `status.extension` before overwriting it.  Otherwise, please add a "todo" comment as @Miciah suggested.

Reply:
I updated the logic to override only `UnmanagedGatewayAPICRDNames` field.